components/higher-order/with-notices/index.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * External dependencies
+ */
+import uuid from 'uuid/v4';
+
+/**
+ * WordPress dependencies
+ */
+import { Component, createHigherOrderComponent } from '@wordpress/element';
+
+/**
+ * Override the default edit UI to include notices if supported.
+ *
+ * @param  {function|Component} BlockEdit Original component.
+ * @return {Component}                    Wrapped component.
+ */
+export default createHigherOrderComponent( ( BlockEdit ) => {
+	return class WrappedBlockEdit extends Component {
+		constructor() {
+			super( ...arguments );
+
+			this.addNotice = this.addNotice.bind( this );
+			this.removeNotice = this.removeNotice.bind( this );
+
+			this.state = {
+				notices: [],
```

Comment:
Wouldn't it be simpler to store notices by ID here?

```js
this.state = {
  1: {},
  2: {},
}
```

That also makes it easier to add and delete.

Reply:
I used an array to follow the same logic we use on the global notices. I wanted to use the same that is being used globally.